pkg/controller/vitesscluster/reconcile_topo.go

```diff
@@ -190,6 +223,17 @@ func (r *ReconcileVitessCluster) reconcileKeyspaceTopology(ctx context.Context,
 		desiredKeyspaces[keyspace.Name] = keyspace
 	}
 
+	if *vt.Spec.TopologyReconciliation.PruneKeyspaces == true {
+		result, err := r.pruneKeyspaces(ctx, vt, ts, desiredKeyspaces, resultBuilder)
+		if err != nil {
+			return result, err
```

Comment:
This should be a `resultBuilder.Merge()` rather than a return. The old code did not bail out if something went wrong with keyspace pruning. It doesn't technically matter right now because `reconcileKeyspaceTopology` is not doing anything else yet, but since it uses `resultBuilder`, we should follow the proper conventions so everything still works as more is added later.

Reply:
Fixed https://github.com/planetscale/vitess-operator/pull/26/commits/03448bb30f936614a4b9582b59fd7bcf24286f8a.